tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+                      [0., 1., 0.],
+                      [0., 0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1., 0.],
+                      [0., 1.]]),
+            np.array([[1.]]),
+            np.array([[1.]]),
+            np.array([[1.]]),
+            np.array([[1.]]),
+            np.array([[1.]]),
+            np.array([[1.]])
+        ]
+
+        blk_eye = BlkDiagMatrix.eye(self.blk_partition)
+        self.allallfunc(blk_eye, result)
+
+    def testBlkDiagMatrixAdd(self):
+        result = [
+            np.array([[-0.5811392 , -0.68058573, -0.09922225,  1.04760434],
+                      [-0.68058573, -0.44429466,  0.38473647, -0.42841058],
+                      [-0.09922225,  0.38473647, -0.77760772, -0.68081541],
+                      [ 1.04760434, -0.42841058, -0.68081541,  0.42896249]]),
+            np.array([[-0.13261882, -0.53132372, -0.85661931],
+                      [-0.53132372, -1.14865325,  0.08966031],
+                      [-0.85661931,  0.08966031, -0.11134927]]),
+            np.array([[-0.13261882, -0.53132372, -0.85661931],
+                      [-0.53132372, -1.14865325,  0.08966031],
+                      [-0.85661931,  0.08966031, -0.11134927]]),
+            np.array([[ 0.0822153 , -0.02080676, -1.1286495 ],
+                      [-0.02080676, -1.20958215, -0.60199042],
+                      [-1.1286495 , -0.60199042,  0.31403212]]),
+            np.array([[ 0.0822153 , -0.02080676, -1.1286495 ],
+                      [-0.02080676, -1.20958215, -0.60199042],
+                      [-1.1286495 , -0.60199042,  0.31403212]]),
+            np.array([[ 0.07894249,  0.43372744],
+                      [ 0.43372744, -0.76793045]]),
+            np.array([[ 0.07894249,  0.43372744],
+                      [ 0.43372744, -0.76793045]]),
+            np.array([[-0.03744825,  0.69332924],
+                      [ 0.69332924, -0.09728449]]),
+            np.array([[-0.03744825,  0.69332924],
+                      [ 0.69332924, -0.09728449]]),
+            np.array([[-0.18610643,  0.74976559],
+                      [ 0.74976559,  0.56202548]]),
+            np.array([[-0.18610643,  0.74976559],
+                      [ 0.74976559,  0.56202548]]),
+            np.array([[-0.31969785]]),
+            np.array([[-0.31969785]]),
+            np.array([[-0.41647978]]),
+            np.array([[-0.41647978]]),
+            np.array([[-0.47055057]]),
+            np.array([[-0.47055057]])
+        ]
+
+        blk_c = self.blk_a + self.blk_b
+        self.allallfunc(result, blk_c)
+
+        blk_c = self.blk_a.add(self.blk_b)
+        self.allallfunc(result, blk_c)
```

Comment:
Right addition?

Reply:
Given two of the same objects, python will invoke lhs operator, so this only comes up for scalars where we have two different object, one of which may not have the method... Operator overloading objects is a little tricky, and ndarrays is subtler yet...  

Regardless b.add(a) is exactly this test up to a,b so it wouldn't add anything here. You are covered both lhs and rhs in scalar add test.

Reply:
Right, nm.